apps/omg/lib/omg/alert/alarm.ex

```diff
@@ -18,13 +18,18 @@ defmodule OMG.Alert.Alarm do
   """
   alias OMG.Alert.AlarmHandler
 
+  @typedoc """
+  The raw alarm being used to `set` the Alarm
+  """
+  @type raw_t :: {:boot_in_progress, node(), module()} | {:ethereum_client_connection, node(), module()}
```

Comment:
this dialyzer warning still occurs
```
apps/omg/lib/omg/alert/alarm.ex:32:contract_subtype
Type specification is a subtype of the success typing.

Function:
OMG.Alert.Alarm.set/1

Type specification:
@spec set(raw_t()) :: :ok | :duplicate

Success typing:
@spec set({:boot_in_progress, _, _} | {:ethereum_client_connection, _, _}) :: :duplicate | :ok
```
```suggestion
  @type raw_t :: {:boot_in_progress | :ethereum_client_connection, any(), any()}
```

to not to get rid of more accurate types we can still change `set`: 
```elixir
def set(raw_alarm = {_, node, reporter}) when is_atom(node) and is_atom(reporter) do
```